tiledb/sm/cpp_api/query_condition.h

```diff
@@ -0,0 +1,225 @@
+  /** Copy constructor. */
+  QueryCondition(const QueryCondition&) = default;
+
+  /** Move constructor. */
+  QueryCondition(QueryCondition&&) = default;
+
+  /** Destructor. */
+  ~QueryCondition() = default;
+
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+
+  /** Copy-assignment operator. */
+  QueryCondition& operator=(const QueryCondition&) = default;
+
+  /** Move-assignment operator. */
+  QueryCondition& operator=(QueryCondition&&) = default;
+
+  /* ********************************* */
+  /*                API                */
+  /* ********************************* */
+
+  /** Returns a shared pointer to the C TileDB query condition object. */
+  std::shared_ptr<tiledb_query_condition_t> ptr() const {
+    return query_condition_;
+  }
+
+  /**
+   * Combines this instance with another instance to form a multi-clause
+   * condition object.
+   *
+   * **Example:**
+   *
+   * @code{.cpp}
+   * int qc1_cmp_value = 10;
+   * tiledb::QueryCondition qc1(
+   *   ctx, "a1", &qc1_cmp_value, sizeof(int), ILEDB_LT);
+   * int qc2_cmp_value = 3;
+   * tiledb::QueryCondition qc2(
+   *   ctx, "a1", &qc2_cmp_value, sizeof(int), TILEDB_GE);
+   *
+   * tiledb::QueryCondition qc3 = qc1.combine(qc2, TILEDB_AND);
+   * query.set_condition(qc3);
+   * @endcode
+   *
+   * @param rhs The right-hand-side query condition object.
+   * @param combination_op The logical combination operator that combines this
+   * instance with `rhs`.
+   */
+  QueryCondition combine(
+      const QueryCondition& rhs,
+      tiledb_query_condition_combination_op_t combination_op) const {
+    auto& ctx = ctx_.get();
+    tiledb_query_condition_t* combined_qc;
+    ctx.handle_error(tiledb_query_condition_combine(
+        ctx.ptr().get(),
+        query_condition_.get(),
+        rhs.ptr().get(),
+        combination_op,
+        &combined_qc));
+
+    return QueryCondition(ctx_, combined_qc);
+  }
+
+  /* ********************************* */
+  /*          STATIC FUNCTIONS         */
+  /* ********************************* */
+
+  /**
+   * Factory function for creating a new query condition with datatype T.
+   *
+   * **Example:**
+   * @code{.cpp}
+   * tiledb::Context ctx;
+   * auto a1 = tiledb::QueryCondition::create<int>(ctx, "a1", 5, TILEDB_LE);
+   * auto a2 = tiledb::QueryCondition::create<std::string>(ctx, "a2", "foo",
+   * TILEDB_NE); auto a3 = tiledb::QueryCondition::create<float>(ctx, "a3", 3.5,
+   * TILEDB_GT); auto a4 = tiledb::QueryCondition::create<double>(ctx,
+   * "a4", 10.0, TILEDB_LT);
+   * @endcode
+   *
+   * @tparam T Datatype of the attribute. Can either be arithmetic type or
+   * string.
+   * @param ctx The TileDB context.
+   * @param name The attribute name.
+   * @param value The value to compare against.
+   * @param op The comparison operator.
+   * @return A new QueryCondition object.
+   */
+  template <typename T>
+  static QueryCondition create(
```

Comment:
Like the constructor can we add overload to handle `std::string`?

Reply:
Done